ec2/ec2test/server.go

```diff
@@ -2536,6 +2539,100 @@ func (srv *Server) modifyInstanceAttribute(w http.ResponseWriter, req *http.Requ
 	}
 }
 
+func (srv *Server) createTags(w http.ResponseWriter, req *http.Request, reqId string) interface{} {
+	var resourceIds []string
+	var tags []ec2.Tag
+
+	for attr, vals := range req.Form {
+		if strings.HasPrefix(attr, "ResourceId.") {
+			fields := strings.SplitN(attr, ".", 2)
+			if len(fields) != 2 {
+				fatalf(400, "InvalidParameterValue", "unknown field %s: %s", attr, vals[0])
+			}
+			i := atoi(fields[1])
+			for i > len(resourceIds) {
+				resourceIds = append(resourceIds, "")
+			}
+			resourceIds[i-1] = vals[0]
+		} else if strings.HasPrefix(attr, "Tag.") {
+			fields := strings.SplitN(attr, ".", 3)
+			if len(fields) != 3 {
+				fatalf(400, "InvalidParameterValue", "unknown field %s: %s", attr, vals[0])
+			}
+			i := atoi(fields[1])
+			for i > len(tags) {
+				tags = append(tags, ec2.Tag{})
```

Comment:
is this right? it will append to tags slice but the slot it ends up in won't necessarily match i

what if we have:

Tag.1.Key
Tag.1.Value
Tag.3.Key
Tag.3.Value

then the tag 3 value will end up in the slot for tag 2.

Is the above possible? Or is it enforced that there must be a tag 1 and tag 2 and tag 3 etc, not missing values like tag 1 and tag 2 and tag4. Either way, I think another test is needed